src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDynamicTests.cs

```diff
@@ -15490,6 +15490,447 @@ class C1
+		IL_0012: ldtoken Program
+		IL_0017: call class [netstandard]System.Type [netstandard]System.Type::GetTypeFromHandle(valuetype [netstandard]System.RuntimeTypeHandle)
+		IL_001c: call class [netstandard]System.Runtime.CompilerServices.CallSiteBinder [Microsoft.CSharp]Microsoft.CSharp.RuntimeBinder.Binder::Convert(valuetype [Microsoft.CSharp]Microsoft.CSharp.RuntimeBinder.CSharpBinderFlags, class [netstandard]System.Type, class [netstandard]System.Type)
+		IL_0021: call class [netstandard]System.Runtime.CompilerServices.CallSite`1<!0> class [netstandard]System.Runtime.CompilerServices.CallSite`1<class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !!T1>>::Create(class [netstandard]System.Runtime.CompilerServices.CallSiteBinder)
+		IL_0026: stsfld class [netstandard]System.Runtime.CompilerServices.CallSite`1<class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !0>> class Program/'<>o__1`1'<!!T1>::'<>p__0'
+		IL_002b: ldsfld class [netstandard]System.Runtime.CompilerServices.CallSite`1<class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !0>> class Program/'<>o__1`1'<!!T1>::'<>p__0'
+		IL_0030: ldfld !0 class [netstandard]System.Runtime.CompilerServices.CallSite`1<class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !!T1>>::Target
+		IL_0035: ldsfld class [netstandard]System.Runtime.CompilerServices.CallSite`1<class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !0>> class Program/'<>o__1`1'<!!T1>::'<>p__0'
+		IL_003a: ldnull
+		IL_003b: callvirt instance !2 class [netstandard]System.Func`3<class [netstandard]System.Runtime.CompilerServices.CallSite, object, !!T1>::Invoke(!0, !1)
+		IL_0040: starg.s b
+		IL_0042: ldarg.0
+		IL_0043: box !!T1
+		IL_0048: ldnull
+		IL_0049: ceq
+		IL_004b: call void [netstandard]System.Console::WriteLine(bool)
+		IL_0050: ret
+	} // end of method Program::'<M1>g__M2|1_0'
+} // end of class Program");
+        }
+
+        [Fact]
+        [WorkItem(41947, "https://github.com/dotnet/roslyn/issues/41947")]
+        public void DynamicInGenericLocalFunction4()
+        {
+            var source = @"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        M1("""");
+        
+        void M1<T>(T a)
+        {
+            M2(a);
+            void M2<T>(T b)
+            {
+                b = (dynamic)default;
+                Console.WriteLine(b is null);
+            }
+        }
+    }
+}";
+            CompileAndVerify(source, expectedOutput: "True", references: new[] { CSharpRef })
+                .VerifyTypeIL("Program", @"
+.class private auto ansi beforefieldinit Program
+    	extends [netstandard]System.Object
+{
+	// Nested Types
+	.class nested private auto ansi abstract sealed beforefieldinit '<>o__0`2'<T, T>
```

Comment:
do we care that these generic type parameters have the same name? I'm guessing it doesn't affect anything except making it harder to interpret the decompiled code?

Reply:
They already have the same name for closures:

https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMACFBGA7AWACgBvIjcrAZi2wDYsAWDAWQgEsA7AHgBUA+ABQBKMhVKEKklgIBEMoQG5RUxi16CeGKABsA9jBESVGccakA3CAgwQMAXgzYlRs+RRNm6gZp37DrsWUAihwAOhwATgEIRSCzAF84ikSXDBT4oA